models/conditional_gans.py

```diff
@@ -65,13 +68,98 @@ class StackGAN1(ConditionalGAN):
     """ Definition for the stage 1 StackGAN """
     def __init__(self, img_size, num_latent_dims, kernel_size,
                  num_filters, reshape_dims):
-        super().__init__(img_size, num_latent_dims, kernel_size,
-                         num_filters, reshape_dims)
+
+        generator = GeneratorStage1(
+            img_size=img_size,
+            num_latent_dims=num_latent_dims,
+            kernel_size=kernel_size,
+            num_filters=num_filters,
+            reshape_dims=reshape_dims
+        )
+
+        discriminator = DiscriminatorStage1(
+            img_size=img_size,
+            kernel_size=kernel_size,
+            num_filters=num_filters
+        )
+
+        super().__init__(
+            generator=generator,
+            discriminator=discriminator,
+            img_size=img_size,
+            num_latent_dims=num_latent_dims,
+            kernel_size=kernel_size,
+            num_filters=num_filters,
+            reshape_dims=reshape_dims
+        )
+
+        self.flatten = Flatten()
+        self.dense_mean = Dense(units=128, activation='relu') # NOTE change units value to variable
+        self.dense_sigma = Dense(units=128, activation='relu') # NOTE change units value to variable
+
+    def generate_conditionals(self, embedding):
+        """ Generate distribution for text embedding.
+            Arguments
+            embedding : Tensor
+                text embedding. Shape (batch_size, feature_size, embedding_size)
+
+            Returns
+            Tensor
+                learnt mean of embedding distribution. Shape (batch_size, reshape_dims/2)
+            Tensor
+                learnt log variance of embedding distribution. Shape (batch_size, reshape_dims/2)
+
+        """
+        x = self.flatten(embedding)
```

Comment:
So we take the mean and var over all elements? Across all samples, all features, and all embedding dimensions?

Reply:
yeah, technically. Might be worth averaging over the sample dimension first then flattening?

Reply:
Yeah we should double check this

Reply:
hard to tell, discussions on the multiple samples are vague. I think there was an older paper that mentions it, will dig it up and see what they do